### ISSUE TITLE
The performance of SmolVLA on LIBERO cannot be replicated

### ISSUE THREAD

**Issue by hahans** [huggingface/lerobot#1369]
I trained SmolVLA from scratch on the LIBERO dataset (the LIBERO dataset under Lerobot), but during the test, I couldn't reproduce its results in the paper. Could there be a problem with my reproduction code or process? Could you produce a version of the reproduction tutorial?

**Comment by zlw21gxy** [comment]
https://github.com/huggingface/lerobot/issues/1316

**Comment by hahans** [comment]
> [#1316](https://github.com/huggingface/lerobot/issues/1316)

Oh, thanks! I saw it. I'd like to use the entire Libero dataset. Have you processed it yet? Alternatively, have there been attempts with other subsets of the Libero dataset?

**Comment by zlw21gxy** [comment]
> > [[#1316](https://github.com/huggingface/lerobot/issues/1316)](https://github.com/huggingface/lerobot/issues/1316)
>
> Oh, thanks! I saw it. I'd like to use the entire Libero dataset. Have you processed it yet? Alternatively, have there been attempts with other subsets of the Libero dataset?

Thanks for the pointer! I came across this repo which shows how to convert the dataset:
[https://github.com/Tavish9/any4lerobot/tree/main/libero2lerobot](https://github.com/Tavish9/any4lerobot/tree/main/libero2lerobot)

However, I haven’t converted the full dataset yet. From what I’ve seen, even after conversion, additional validation (like move checks) is likely needed, since there have been version and definition changes across releases.

It’s a bit surprising there isn’t a comprehensive, end-to-end example showing how to convert, validate, and reproduce results with the entire Libero dataset—or maybe I’ve missed one? Would appreciate any pointers if such a reference exists.

**Comment by JustinKai0527** [comment]
@zlw21gxy  thanks for providing code, I have few question about how to adapt to smolvla in libero, 
like     
max_action_dim: int = 32 in libero we just output dim=7
and for SmolVLAPolicy it need dataset_stats for init() 
if I want to do zero-shot without finetuning how to get the dataset_stats, thx

**Comment by jadechoghari** [comment]
We officially ported LIBERO.
You can use this checkpoint to evaluate [HuggingFaceVLA/smolvla_libero](https://huggingface.co/HuggingFaceVLA/smolvla_libero) (with success rate between 80 and 100) and the doc below https://huggingface.co/docs/lerobot/en/libero
Thank you.

**Comment by jadechoghari** [comment]
This issue has been addressed, so I’ll close it for now.
Please feel free to reopen if needed.

**Comment by 999Tom** [comment]
Hello @jadechoghari in your repo [HuggingFaceVLA/smolvla_libero](https://huggingface.co/HuggingFaceVLA/smolvla_libero), the policy used is ACT but not smolVLA. Also, there seems no libero used. Could u please further explain it? Thanks in advance

**Comment by jadechoghari** [comment]
Hey! There’s no ACT here in the ckpt, where do you see that? 😄 The policy used is SmolVLA (see the type field here: [config.json](https://huggingface.co/HuggingFaceVLA/smolvla_libero/blob/main/config.json)).
We trained it on the [LIBERO dataset](https://huggingface.co/datasets/HuggingFaceVLA/libero), and you can easily evaluate it in LIBERO by following the guide in our [docs](https://huggingface.co/docs/lerobot/en/libero).

**Comment by 999Tom** [comment]
OH, I saw it on the ModelCard example. But it is okay, I understand it now. Thank you very much >,<

**Comment by zhoutao-sdu2025** [comment]
> 我们正式移植了 LIBERO。您可以使用此检查点来评估 [HuggingFaceVLA/smolvla_libero](https://huggingface.co/HuggingFaceVLA/smolvla_libero)（成功率在 80 到 100 之间）和下面的文档 https://huggingface.co/docs/lerobot/en/libero 谢谢。

Hello, I conducted evaluation and training on the 5090 according to your tutorial, but the success rate was 0, which puzzles me.

<img width="2442" height="167" alt="Image" src="https://github.com/user-attachments/assets/4ca1d1de-ce45-4eab-aba6-46063fbf967f" />

**Comment by zhoutao-sdu2025** [comment]
> Hey! There’s no ACT here in the ckpt, where do you see that? 😄 The policy used is SmolVLA (see the type field here: [config.json](https://huggingface.co/HuggingFaceVLA/smolvla_libero/blob/main/config.json)). We trained it on the [LIBERO dataset](https://huggingface.co/datasets/HuggingFaceVLA/libero), and you can easily evaluate it in LIBERO by following the guide in our [docs](https://huggingface.co/docs/lerobot/en/libero).

The instructions I use are as follows:    lerobot-eval   --policy.path=/home/sdu/smolvla_model/100000/pretrained_model_migrated --env.type=libero  --env.task=libero_object   --eval.batch_size=2   --eval.n_episodes=3  

<img width="2417" height="1041" alt="Image" src="https://github.com/user-attachments/assets/11851a5b-c37d-4eca-9173-3bbe433bdd49" />

**Comment by jadechoghari** [comment]
Hi, can you share your checkpoint if it is on the hub

**Comment by zhoutao-sdu2025** [comment]
> Hi, can you share your checkpoint if it is on the hub

I'm sorry,  our school has set up protection, so we can't access Hugging Face. That's why I downloaded the corresponding model and dataset offline.

<img width="266" height="249" alt="Image" src="https://github.com/user-attachments/assets/77148674-b21a-4bfa-b6e8-cc9900e260bf" />


<img width="246" height="215" alt="Image" src="https://github.com/user-attachments/assets/d24d4903-3c23-481c-85fb-c395b009565c" />

<img width="369" height="943" alt="Image" src="https://github.com/user-attachments/assets/3dea7660-5595-4e1d-87e1-22409583f809" />

<img width="732" height="268" alt="Image" src="https://github.com/user-attachments/assets/0f6b15af-ab73-4ef5-bfe9-2747bdae8474" />

I haven't changed any internal code.  I used 5090 for training, and I don't know if it has any impact.

**Comment by jadechoghari** [comment]
Ok thank you, we are investigating the issue.
Can you confirm you're having high success rate with this ckpt: https://huggingface.co/HuggingFaceVLA/smolvla_libero 
Tip: It is best to use mujoco 3.3.2, as older version can have different colors rendered.
https://github.com/Lifelong-Robot-Learning/LIBERO/issues/88

**Comment by zhoutao-sdu2025** [comment]
> Ok thank you, we are investigating the issue. Can you confirm you're having high success rate with this ckpt: https://huggingface.co/HuggingFaceVLA/smolvla_libero Tip: It is best to use mujoco 3.3.2, as older version can have different colors rendered. [Lifelong-Robot-Learning/LIBERO#88](https://github.com/Lifelong-Robot-Learning/LIBERO/issues/88)

Hello, I am not using this ckpt: https://huggingface.co/HuggingFaceVLA/smolvla_libero, but rather https://huggingface.co/HuggingFaceVLA/smolvla_libero_ckpts/100000. I would like to conduct an evaluation. I downloaded version 3.3.2 of mujoco, but the success rate has not improved.

<img width="1961" height="332" alt="Image" src="https://github.com/user-attachments/assets/cc79066f-ebdd-4f39-9e8f-fd3e66a8c43b" />

**Comment by jadechoghari** [comment]
Those ckpts are depreciated (before pipeline breaking changes and won't work). You should use the one mentioned above :)

**Comment by zhoutao-bjd** [comment]
> Those ckpts are depreciated (before pipeline breaking changes and won't work). You should use the one mentioned above :)

Okay, I'd like to ask how many training steps are approximately needed to increase the success rate when using https://huggingface.co/HuggingFaceVLA/smolvla_libero. The dataset I'm using is https://huggingface.co/datasets/HuggingFaceVLA/libero, and the code I'm using is as follows: lerobot-train --policy.type=smolvla --policy.repo_id=/home/sdu/smolvla_model --dataset.repo_id=/home/sdu/libero --env.type=libero --env.task=libero_spatial --output_dir=./outputs/ --steps=100000 --batch_size=8 --eval.batch_size=1 --eval.n_episodes=1 --eval_freq=1000 \

<img width="2423" height="899" alt="Image" src="https://github.com/user-attachments/assets/d552fd03-68fc-439b-a28e-69b79e7ec21a" />

**Comment by zhoutao-sdu2025** [comment]
> Those ckpts are depreciated (before pipeline breaking changes and won't work). You should use the one mentioned above :)

Hello, teacher. When I was conducting the test, I trained for 100k steps, but the highest accuracy during training evaluation was 0.6, while during single evaluation it was around 0.5 at most. I don't know why, and I'd like to know how many steps you usually train for.

**Comment by zhoutao-sdu2025** [comment]
> Those ckpts are depreciated (before pipeline breaking changes and won't work). You should use the one mentioned above :)

Hello，When I trained to around 60k, the loss has been stuck at 0.09, and I used your original code.
  CUDA_VISIBLE_DEVICES=0 lerobot-train   --policy.type=smolvla   --policy.repo_id=/home/sdu/smolvla_model   --dataset.repo_id=/home/sdu/libero  --env.type=libero   --env.task=libero_object   --output_dir=./outputs/   --policy.load_vlm_weights=true  --steps=100000   --batch_size=4   --eval.batch_size=1   --eval.n_episodes=1   --eval_freq=1000 \

This is the result of running 100k steps, and the test only has 30% accuracy.
<img width="2407" height="178" alt="Image" src="https://github.com/user-attachments/assets/efe3cdf7-a4df-460a-bd37-35966ebdd6f9" />

**Comment by zimgong** [comment]
Hi, a quick follow-up, I was able to replicate the results with this model: https://huggingface.co/HuggingFaceVLA/smolvla_libero. However, I cannot achieve similar performance with my own model following #2107. Curious for a solution, thanks!

**Comment by jadechoghari** [comment]
Hello @zimgong @zhoutao-bjd we had an issue with the new converted LIBERO dataset v2.1 to v3.0 
This should now be solved, as we used it to train and evaluate on our ported version of pi0.5
So I would suggested deleting the dataset locally, downloading it again and launching training !

**Comment by zimgong** [comment]
Hi @jadechoghari, I used the latest v3.0 LIBERO dataset, and the results weren't comparable. I wonder if there is any additional training tricks we missed?

**Comment by zhoutao-sdu2025** [comment]
> 您好，我们在新转换的 LIBERO 数据集 v2.1 到 v3.0 时遇到了问题，现在应该解决了这个问题，因为我们用它来训练和评估移植的 pi0.5 版本所以我建议在本地删除数据集，重新下载并启动训练！

Thank you very much for your helpn. I have almost solved my problem. I increased the batch-size and set it to 32. After that, the loss dropped to 0.7, with the accuracy ranging between 60% and 100%, and the highest accuracy during evaluation was 92%. I will try new datasets and a batch size of 128 or 256 later.

**Comment by zhoutao-sdu2025** [comment]
> 嗨，我使用了最新的 v3.0 LIBERO 数据集，结果没有可比性。不知道还有没有我们遗漏的额外训练技巧？

You can try increasing the batch size. Initially, I set it to 4 or 8 with poor results; the larger, the better.

**Comment by zimgong** [comment]
Hi @zhoutao-bjd . I'm using a batch size of 128 right now on a single 4090 tbh. I wonder did you train for all the 100k steps? I found the train loss to plateau around 30k. It could also be that I'm training a 0.24B model, which is less capable by nature.

**Comment by jadechoghari** [comment]
@zimgong Can you completely delete the dataset and download it again? We’ve introduced some modifications to it:
https://huggingface.co/datasets/HuggingFaceVLA/libero

**Comment by zimgong** [comment]
@jadechoghari Yeah I redownloaded and benchmarked both the 0.45B and 0.24B models. The 0.45B was less competent than the official, but was good enough. I only tested for 4 trials each task so there should be some sort of deviation. However, the 0.24B model was not working that well. I wonder if there's training tips for the 0.24B model? Like shall we unfreeze the VLM backbone? 

<img width="999" height="291" alt="Image" src="https://github.com/user-attachments/assets/da54004c-a752-4bdf-b97b-58be233c9c6d" />

**Comment by jadechoghari** [comment]
Hi @zimgong this is a closed issue, let's talk here https://github.com/huggingface/lerobot/issues/2107
